### PR TITLE
Mod id in panel

### PR DESCRIPTION
**Pull request by LordMidas**

### CODE REVIEW

File: msu/systems/mod_settings/abstract_setting.nut [hunk 1]
```diff
@@ -160,25 +160,28 @@
 	}
 
 
-	function flagSerialize( _modID )
+	function flagSerialize()
 	{
-		this.setFlagForProperty("Value", _modID);
-		this.setFlagForProperty("Locked", _modID);
-		this.setFlagForProperty("LockReason"_modID);
+		local modID = this.getModID();
```
Comment (Enduriel) [inline msu/systems/mod_settings/abstract_setting.nut:165]: I wouldn't bother with a local variable when the function is this short

File: msu/systems/mod_settings/abstract_setting.nut [hunk 2]
```diff
@@ -160,25 +160,28 @@
...
+		this.setFlagForProperty("LockReason"modID);
 	}
 
-	function flagDeserialize( _modID )
+	function flagDeserialize()
 	{
-		this.setPropertyIfFlagExists("Value", _modID);
-		this.setPropertyIfFlagExists("Locked", _modID);
-		this.setPropertyIfFlagExists("LockReason", _modID);
+		local modID = this.getModID();
```
Comment (Enduriel) [inline msu/systems/mod_settings/abstract_setting.nut:173]: as above

File: msu/systems/mod_settings/abstract_setting.nut [hunk 3]
```diff
@@ -160,25 +160,28 @@
...
+		this.setPropertyIfFlagExists("LockReason", modID);
 	}
 
-	function resetFlags( _modID )
+	function resetFlags()
 	{
-		this.clearFlagForProperty("Value", _modID);
-		this.clearFlagForProperty("Locked", _modID);
-		this.clearFlagForProperty("LockReason", _modID);
+		local modID = this.getModID();
```
Comment (Enduriel) [inline msu/systems/mod_settings/abstract_setting.nut:181]: as above

File: msu/systems/mod_settings/settings_element.nut [hunk 4]
```diff
@@ -40,6 +40,11 @@
 		return this.getPage().getPanelID();
 	}
 
+	function getModID()
+	{
+		return this.getPage().getPanel().getModID();
```
Comment (Enduriel) [inline msu/systems/mod_settings/settings_element.nut:45]: Shouldn't this be simplifies to `this.getPage.getModID()`?

**Review by Enduriel — CHANGES_REQUESTED** [review]

**Review by TaroEld — COMMENTED** [review]
Signing Enduriels comments

File: msu/systems/mod_settings/settings_element.nut (hunk 4 above)
Comment (LordMidas) [reply msu/systems/mod_settings/settings_element.nut:45]: Will do

File: msu/systems/mod_settings/abstract_setting.nut (hunk 1 above)
Comment (LordMidas) [reply msu/systems/mod_settings/abstract_setting.nut:165]: It's a single function call vs 3 function calls to a function which calls other functions, so why is it not preferable?

**Pull request merged**